tasks/views.py

```diff
@@ -15,8 +15,23 @@ def view_tasks(request):
 
     if request.method == "POST":
         priority = request.POST.get('priority')
+        status = request.POST.get('status')
+
+        if status == 'InProgress':
+            status = 'in_progress'
+
+        if status == "" and priority == "":
+            return render(request, 'tasks/tasks.html', context)
+
+        if status == "":
+            context['tasks'] = context['tasks'].filter(priority=Priority[priority.upper()])
+            return render(request, 'tasks/tasks.html', context)
+
         if priority == "":
+            context['tasks'] = context['tasks'].filter(status=Status[status.upper()])
             return render(request, 'tasks/tasks.html', context)
+
+        context['tasks'] = context['tasks'].filter(status=Status[status.upper()])
         context['tasks'] = context['tasks'].filter(priority=Priority[priority.upper()])
 
     return render(request, 'tasks/tasks.html', context)
```

Comment:
I can see that you repeat the same return state a few times maybe we can do it something like:
```suggestion
        if status == "" and priority != "":
            context['tasks'] = context['tasks'].filter(priority=Priority[priority.upper()])
        elif priority == "" and status != "":
            context['tasks'] = context['tasks'].filter(status=Status[status.upper()])
        elif priority != "" and status != "":
             context['tasks'] = context['tasks'].filter(status=Status[status.upper()])
             context['tasks'] = context['tasks'].filter(priority=Priority[priority.upper()])
    return render(request, 'tasks/tasks.html', context)
```